src/testdir/test_quickfix.vim

```diff
@@ -1085,6 +1085,16 @@ func Test_efm2()
   call assert_equal(1, l[4].valid)
   call assert_equal('unittests/dbfacadeTest.py', bufname(l[4].bufnr))
 
+  " Test for %o
+  set efm=%f(%o):%l\ %m
+  cgetexpr ['src/Language/PureScript/Types.hs(Language.PureScript.Types):20 Error']
+  let l = getqflist()
+  call assert_equal(1, len(l), string(l))
+  call assert_equal('Language.PureScript.Types', l[0].module)
+  copen
```

Comment:
You may want to add a test to jump to the entry to make sure that the filename
is used.